### PR TITLE
849 update pollsubgraph

### PR DESCRIPTION
**Pull request by wanglonghong**
## Checklist

- [ ] Test manually on `test-ui` using remote chains.
- [ ] Run load tests.
- [ ] Update documentation if needed.
- [ ] Update CHANGELOG.md.

### CODE REVIEW

File: packages/agents/router/src/config.ts [hunk 1]
```diff
@@ -166,8 +166,8 @@ export const getEnvConfig = (
 
     const maxLag = chainConfig.subgraph?.maxLag ?? MIN_SUBGRAPH_SYNC_BUFFER;
     nxtpConfig.chains[domainId].subgraph = {
-      runtime: chainConfig.subgraph?.runtime ?? chainDataForChain?.subgraph ?? [],
-      analytics: chainConfig.subgraph?.analytics ?? chainDataForChain?.analyticsSubgraph ?? [],
+      runtime: chainConfig.subgraph?.runtime ?? chainDataForChain?.subgraphs.runtime ?? [],
```
Comment (jakekidd) [inline packages/agents/router/src/config.ts:169]: have we updated `chainData.json` to reflect this?

Comment (wanglonghong) [reply packages/agents/router/src/config.ts:180]: yes. we updated it in the repository. connext/chainData

File: packages/adapters/subgraph/src/reader.ts [hunk 2]
```diff
@@ -171,23 +177,31 @@ export class SubgraphReader {
       [...txIdsByDestinationDomain.entries()].map(async ([destinationDomain, transferIds]) => {
         const subgraph = this.subgraphs.get(destinationDomain)!; // should exist bc of initial filter
 
-        await subgraph.runtime.request<GetExecutedAndReconciledTransfersByIdsQuery>(
+        const { transfers } = await subgraph.runtime.request<GetExecutedAndReconciledTransfersByIdsQuery>(
           (client) =>
             client.GetExecutedAndReconciledTransfersByIds({
               transferIds,
-              maxXCalledBlockNumber: agents.get(destinationDomain)!.maxXCallBlockNumber.toString(),
+              maxXCalledBlockNumber: agents.get(destinationDomain)!.maxBlockNumber.toString(),
```
Comment (jakekidd) [inline packages/adapters/subgraph/src/reader.ts:184]: As far as I can tell, we are no longer using this param in `GetExecutedAndReconciledTransfersByIds`? It looks like it's been removed from the query

Comment (jakekidd) [reply packages/adapters/subgraph/src/reader.ts:184]: Although, we might want to add it back in? Unsure

**Review by rhlsthrm — APPROVED** [review]

File: ops/testnet/config.tf [hunk 3]
```diff
@@ -4,24 +4,40 @@ locals {
...
-        "subgraph" = {
-          "runtime" = ["https://api.thegraph.com/subgraphs/name/connext/nxtp-amarok-runtime-v0-kovan"]
-          "analytics" : [""]
+        providers = ["https://eth-kovan.alchemyapi.io/v2/${var.kovan_alchemy_key_0}"]
+        subgraph = {
+          runtime = [{
+            query = "https://api.thegraph.com/subgraphs/name/connext/nxtp-amarok-runtime-v0-kovan"
+            health = "https://api.thegraph.com/index-node/graphql"
+          }]
+          analytics = [
```
[the remark this answers is not in the document]
Comment (wanglonghong) [reply ops/testnet/config.tf:35]: makes sense

File: ops/testnet/config.tf [hunk 4]
```diff
@@ -49,9 +65,13 @@ locals {
...
         subgraph = {
-          runtime = ["https://api.thegraph.com/subgraphs/name/connext/nxtp-amarok-runtime-v0-rinkeby"]
-          analytics : [""]
+          runtime = [{
+              query = "https://api.thegraph.com/subgraphs/name/connext/nxtp-amarok-runtime-v0-rinkeby"
+              health = "https://api.thegraph.com/index-node/graphql"
+          }]
+          analytics: []
         }
+        deployments = {}
```
[the remark this answers is not in the document]
Comment (wanglonghong) [reply ops/testnet/config.tf:74]: makes sense

File: packages/adapters/subgraph/src/lib/subgraphs/runtime/queries.ts [hunk 5]
```diff
@@ -187,16 +187,8 @@ export const getTransfer = gql`
...
-    transfers(
-      where: {
-        transferId_in: $transferIds
-        xcalledBlockNumber_lte: $maxXCalledBlockNumber
-        status_in: [Executed, Reconciled]
-      }
-      orderBy: xcalledBlockNumber
-      orderDirection: desc
-    ) {
+  query GetExecutedAndReconciledTransfersByIds($transferIds: [Bytes!]) {
```
[the remark this answers is not in the document]
Comment (wanglonghong) [reply packages/adapters/subgraph/src/lib/subgraphs/runtime/queries.ts:190]: yes. right. its difficult to decide if the status is either Executed or Reconciled before querying by Ids.
Any better idea???

File: packages/agents/router/src/config.ts [hunk 6]
```diff
@@ -173,8 +177,8 @@ export const getEnvConfig = (
...
```
[the remark this answers is not in the document]
Comment (wanglonghong) [reply packages/agents/router/src/config.ts:180]: good catch. will do

File: packages/adapters/subgraph/src/lib/subgraphs/runtime/queries.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (wanglonghong) [reply packages/adapters/subgraph/src/lib/subgraphs/runtime/queries.ts]: :)

[the remark this answers is not in the document]
Comment (rhlsthrm) [reply packages/adapters/subgraph/src/lib/subgraphs/runtime/queries.ts]: Let's leave it for now as a TODO in the code.

[the remark this answers is not in the document]
Comment (rhlsthrm) [reply packages/adapters/subgraph/src/lib/subgraphs/runtime/queries.ts]: Oh lol you already did it. Nice!

[the remark this answers is not in the document]
Comment (wanglonghong) [reply packages/adapters/subgraph/src/lib/subgraphs/runtime/queries.ts]: Yes, I don't like TODO. lol

**Review by rhlsthrm — APPROVED** [review]

**Pull request merged**